livox_ros_driver/CMakeLists.txt

```diff
@@ -199,12 +211,58 @@ target_link_libraries(${PROJECT_NAME}_node
 	${APR_LIBRARIES}
 	)
 
+#---------------------------------------------------------------------------------------
+# ADD NODELET LIBRARIES
+#---------------------------------------------------------------------------------------
+include_directories(
+    ${catkin_INCLUDE_DIRS}
+    ${PCL_INCLUDE_DIRS}
+    ${APR_INCLUDE_DIRS}
+    common
+    common/rapidjson
+    comon/rapdidxml
+    common/comm
+    timesync
+    timesync/user_uart
+    livox_ros_driver
+	${CMAKE_CURRENT_SOURCE_DIR}/Livox-SDK/sdk_core/include
+    )
+
+add_library(${PROJECT_NAME}_nodelet
+	livox_ros_driver/livox_ros_driver_nodelet.cpp
+	livox_ros_driver/lvx_file.cpp
+	livox_ros_driver/ldq.cpp
+	livox_ros_driver/lds.cpp
+	livox_ros_driver/lds_lvx.cpp
+	livox_ros_driver/lds_lidar.cpp
+	livox_ros_driver/lds_hub.cpp
+	livox_ros_driver/lddc.cpp
+	livox_ros_driver/livox_ros_driver.cpp
+	timesync/timesync.cpp
+	timesync/user_uart/user_uart.cpp
+	common/comm/comm_protocol.cpp
+	common/comm/sdk_protocol.cpp
+	common/comm/gps_protocol.cpp
+	)
+
+target_link_libraries(${PROJECT_NAME}_nodelet
+	# livox_sdk_static.a
```

Comment:
Is this supposed to be commented out? I think that may explain why the nodelet couldn't find the `GetLivoxSdkVersion` symbol.

Reply:
That's weird, I don't understand why it didn't fail for me. I'll uncomment this and test with a clean workspace build.

Reply:
Beautiful error:

```
/usr/bin/ld: /usr/local/lib/liblivox_sdk_static.a(device_discovery.cpp.o): relocation R_X86_64_TPOFF32 against symbol `_ZGVZN6spdlog7details2os9thread_idEvE3tid' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: /usr/local/lib/liblivox_sdk_static.a(livox_sdk.cpp.o): relocation R_X86_64_PC32 against symbol `_ZN3fmt2v58internal12basic_bufferIcE6resizeEm' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
livox_ros_driver/livox_ros_driver/CMakeFiles/livox_ros_driver_nodelet.dir/build.make:748: recipe for target '/home/borghi/ws_core/devel/lib/liblivox_ros_driver_nodelet.so' failed
```

Reply:
If I keep this line commented out, it works for me:

```
$ roslaunch lidar_algorithms livox.launch

[ INFO] [1653327915.913936298]: Livox Ros Driver Version: 2.6.0
[ INFO] [1653327915.916519360]: Data Source is raw lidar.
[ INFO] [1653327915.916750185]: Config file : /home/borghi/ws_core/src/livox_ros_driver/livox_ros_driver/config/livox_lidar_config.json
Commandline input bd:100000000000000
Invalid bd:100000000000000!
Livox SDK version 2.3.0
broadcast code[1PQDH5B00100041] : 0 0 0 0 0 0
broadcast code[0TFDG3U99101431] : 0 0 0 0 0 0
Disable timesync
No broadcast code was added to whitelist, swith to automatic connection mode!
Livox-SDK init success!
```

Reply:
I'm thinking that we'll probably want to catkinize the Livox-SDK package so that it's correctly built. It seems like you have a previously installed version in `/usr/local/lib` that's conflicting with the version built in `~/catkin_ws/src/livox_ros_driver/Livox-SDK`.

Reply:
It took me a while to compile the whole image again, but I tested with the latest commit and it worked.

```
$ roslaunch lidar_algorithms livox.launch

[ INFO] [1653339923.976673677]: Livox Ros Driver Version: 2.6.0
[ INFO] [1653339923.978957731]: Data Source is raw lidar.
[ INFO] [1653339923.979294460]: Config file : /home/borghi/ws_core/src/livox_ros_driver/livox_ros_driver/config/livox_lidar_config.json
Commandline input bd:100000000000000
Invalid bd:100000000000000!
Livox SDK version 2.3.0
broadcast code[1PQDH5B00100041] : 0 0 0 0 0 0
broadcast code[0TFDG3U99101431] : 0 0 0 0 0 0
Disable timesync
No broadcast code was added to whitelist, swith to automatic connection mode!
Livox-SDK init success!
[ INFO] [1653339923.981091665]: Init lds lidar success!
```

Reply:
For the record, the nodelet manager and its node:

![Screenshot from 2022-05-23 18-06-50](https://user-images.githubusercontent.com/3400230/169906208-de5eb094-6c92-4135-9e00-2ce38052705c.png)

Reply:
I need to push some changes to `oxin_brains`.

Reply:
Well, the PR was already done: https://github.com/oxin-ros/oxin_brains/pull/170